### PR TITLE
Fixes #1950: counter for $

### LINKED ISSUE #1950: {counter}$ is not work
the counter value is overlooked

### PR DESCRIPTION
**Pull request by tagniam**
Implements counter for line end actions(`$`, `<End>`, `<D-right>`). Fixes #1950

### CODE REVIEW

**Comment by Chillee** [comment]
@tagniam Could you write some unit tests covering this case?

**Comment by tagniam** [comment]
@Chillee done!

File: src/actions/motion.ts [hunk 1]
```diff
@@ -740,12 +740,16 @@ class MoveRepeatReversed extends BaseMovement {
 }
 
 @RegisterAction
-class MoveLineEnd extends BaseMovement {
+class MoveDownLineEnd extends BaseMovement {
```
Comment (Chillee) [inline src/actions/motion.ts]: I think it's better to keep this as `MoveLineEnd`. That's the functionality that most people associate with this key.

**Comment by Chillee** [comment]
Thanks for your work!

**Pull request merged**